### ISSUE TITLE
Figure out the ownership of SB secret

### ISSUE THREAD

**Issue by metacosm** [snowdrop/component-operator#76]
Right now, the capabilty owns the secret but maybe it is already own by the DB resource.

**Comment by cmoulliard** [comment]
Even if I don't like the idea, we must perhaps duplicate the `Secret` to keep one owned by the KubeDB operator and another by the `Component` as this secret represents the info needed to bind the DB parameters with the Datasource, JDBC connection string, ... 

Our strategy here will depend if we delete the KubeDB CRD when we delete a Capability. WDYT ? @metacosm

**Comment by cmoulliard** [comment]
I think that the question about the ownership of the secret could not be discussed alone without taking care about the `Link CRD`. 

When we were using the service catalog,  a `Link CRD` was needed to bind from a secret the ENV vars to a Deployment = Component = Microservices. Such Env vars were next used by the Spring boot Application to create the Datasource. Then I think that we should keep this approach and that we do the following

**Creation step**
- When a capability will create a DB, then a DB's secret is created and will be owned by the KubeDB CRD/operator
- Secret is created as defined within the ticket #75 
- Secret is used by KubeDB to get the USER, PWD or Database name to be created
- DB is created within current namespace

** Binding step**
- The secret of the `DB_CONFIG_NAME` is used to create a `Link` to the `Component` where we would like to inject ENV vars from the DB
- A Link CRD is created
- A Secret is generated using the DB's secret where secret name = `DB_CONFIG_NAME`
- The secret is owned by the Component
- Deployment is modified to inject the ENVs using ENV From Source => secret

**Deletion Step**
- When a capability is deleted, then the KubeDB owned resources will be deleted + DB's secret
- When a `Component` is deleted, then we delete all its owned resources + bind secret

Make sense - @metacosm ?

**Comment by cmoulliard** [comment]
Should we keep this ticket open ? @metacosm

**Comment by metacosm** [comment]
Yes, this is still something we need to address in particular with the new capability mechanism.

**Comment by cmoulliard** [comment]
Do you plan to work on this task this week ? When will it be delivered as part of a next release ? @metacosm
